asyncscope.md

```diff
@@ -152,32 +154,52 @@ struct async_scope {
     async_scope& operator=(const async_scope&) = delete;
     async_scope& operator=(async_scope&&) = delete;
 
+    auto nest(sender) const&->nested_sender<Values…>;
+
     auto spawn(sender) const&->void;
     auto spawn_future(sender) const&->future_sender<Values…>;
 
-    auto empty() const&->empty_sender;
+    auto on_empty() const&->empty_sender;
 
     auto get_stop_source() & ->stop_source&;
     auto get_stop_token() const& ->stop_token;
     auto request_stop() & ->void;
 };
 
+template<class... Ts>
+struct nested_sender; // see below;
 
 template<class... Ts>
 struct future_sender; // see below;
+
+struct empty_sender; // see below;
 ```
 
 ## Lifetime
 
 An `async_scope` object must outlive work that is spawned on it. It should be viewed as owning the storage for that work.
 The `async_scope` may be constructed in a local context, matching syntactic scope or the lifetime of surrounding algorithms.
-The destructor of the `async_scope` will terminate if there is outstanding work in the scope at destruction time, therefore `empty` **must** be called before the `async_scope` object is destroyed.
+The destructor of an `async_scope` will terminate if there is outstanding work in the scope at destruction time, therefore `on_empty` **must** be called and the returned `empty_sender` **must** have started and completed with `void`, before the `async_scope` object is destroyed.
+
+## nest
+
+`nest(sender) const&->nested_sender<Values…>;`
+
+returns a `nested_sender` that extends the lifetime of the `async_scope` that produced it. 
```

Comment:
Does this extend the lifetime of the scope, or extend the lifetime of the sender to that of the scope? You are nesting a sender within the scope, right? So I think the latter. When I first read what you wrote here I thought this was a way of nesting the entire scope inside another sender, but we already have on_empty that does that.

Reply:
yes, the `nested_sender` extends the scope of the `async_scope` to include its own object lifetime and, if started, the lifetime of the sender given to `nest()`